### DISCUSSION TITLE
clusterNetworks ip ranges

### DISCUSSION THREAD

**Discussion by DaspawnW** [linkerd/linkerd2#6457] [Q&A]
Based on this issue: https://github.com/linkerd/linkerd2/issues/5852 
and the comments inside it: https://github.com/linkerd/linkerd2/issues/5852#issuecomment-795670544
I now double checked my clusterNetwork to be save about the configuration.

Therefore the following question:
Does the clusterNetwork only includes the Pod CIDR Range or does it also include the Service IP CIDR Range?
Our current configuration is as follows:

clusterNetworks: 10.32.0.0/22,10.200.0.0/16

Where 10.32.0.0/22 the Service IP CIDR Range is and 10.200.0.0/16 the IP range for Pods.

Thank you in advance :-)

**Comment by olix0r** [comment]
@DaspawnW That's correct: `clusterNetworks` needs to cover all IPs for which the proxy should do discovery (for load balancing and endpoint metadata, including mTLS).

Could we make this clearer in the docs? Where would you want to see this?

**Reply by DaspawnW** [reply]
Hi @olix0r,
thanks for the fast reply :-) It's always a pleasure how fast Linkerd2 community and team is

I struggled based on the helm chart [values.yaml](https://github.com/linkerd/linkerd2/blob/ba21a970eb316eba110e2d17bea64c063dd67725/charts/linkerd2/values.yaml#L8): The cluster networks for which service discovery is performed. This should include the pod network but need not include the node network.

Maybe I simply missunderstood, but for me I came over the pod network and thought it could only be the ip range of pods.

**Reply by kleimkuhler** [reply]
Thanks for the feedback @DaspawnW. #6489 clarifies that pod _and_ service networks should be in the cluster networks value. Would this have helped answer your original question?
